### ISSUE TITLE
Extending `vue/order-in-components` to accept order for elements inside `order` for eg. setup

### ISSUE THREAD

**Issue by hir06** [vuejs/eslint-plugin-vue#2055]
I am using a composition API and setup() does heavy lifting for the component. having methods, lifecycle hooks, return defined inside it.
while multiple people working on the project we try to enforce the order and get it in sync at the time of PR review.
However I see there's a rule which helps defining the order inside components. 
can we extend that for the `setup` function to follow the specific order

**Please describe what the rule should do:**
enforce order inside setup function in composition API
for eg.
 ```
'vue/order-in-components': [
  'error',
  {
    order: [
      'el',
      'setup': [
        'methods',
        ['template', 'render'],
        'LIFECYCLE_HOOKS',
        'return',
      ],
    ],
  },
],
```

**What category should the rule belong to?**

- [ ] Enforces code style (layout)

**Provide 2-3 code examples that this rule should warn about:**

```vue
setup() {
  onBeforeMount() { }
  showDialog() { }
  return {
  }
}

Error: Methods should appear before LIFECYCLE_HOOKS
```

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for posting the issue.

But I don't think it helps to force order in the Composition API.

The Composition API features more flexible code organization, as shown in the following documentation:
https://vuejs.org/guide/extras/composition-api-faq.html#more-flexible-code-organization

I think enforcing order in the Composition API goes against this feature.
